### PR TITLE
test tha produces an error for decoding incarnation

### PR DESCRIPTION
**Pull request by enriavil1**

### CODE REVIEW

File: core/types/accounts/account_test.go [hunk 1]
```diff
@@ -277,6 +278,33 @@ func testIncarnationWithNoIncarnation(t *testing.T) {
 
 }
 
+func testIncarnationWithInvalidEncodedAccount(t *testing.T) {
```
Comment (AlexeyAkhunov) [inline core/types/accounts/account_test.go:281]: I think if it starts with the lowercase letter, it does not automatically get executed, please check. I think it needs to start with capital "T"

**Comment by AskAlexSharov** [comment]
In other tests also need check error instead of `_ = DecodeIncarnationFromStorage`

**Comment by enriavil1** [comment]
> In other tests also need check error instead of `_ = DecodeIncarnationFromStorage`

I check for errors previously using the encodedAccount. If it has no error then I run it again and test if its equal to the original incarnation.

**Review by Giulio2002 — APPROVED** [review]
Now it should be fine

**Comment by AskAlexSharov** [comment]
> @AskAlexSharov does it looks fine now?

I still see unchecked errors: `_ = DecodeIncarnationFromStorage`

**Comment by enriavil1** [comment]
@AskAlexSharov sorry, but I checked for errors in the lines before. Should I just check on this line too?

**Comment by enriavil1** [comment]
I have added the error check in those lines

**Comment by AskAlexSharov** [comment]
> @AskAlexSharov sorry, but I checked for errors in the lines before. Should I just check on this line too?

If method returns error - then need check it. 

Imagine I in future will add one more validation rule to method and your test will not catch it - because error was lost.

**Pull request merged**